u8timeseries/utils/timeseries_generation.py

```diff
@@ -0,0 +1,137 @@
+    :param y_offset: The shift of the sine function along the y axis.
+    :param length: The length of the returned TimeSeries.
+    :param offset: The time differene between two adjacent entries in the returned TimeSeries. A DateOffset alias is expected.
+    :param start_date: The time index of the first entry in the returned TimeSeries.
+    :return: A sinusoidal TimeSeries parametrized as indicated above.
+    """
+
+    times = pd.date_range(periods=length, freq=offset, start=start_date)
+    values = np.array(range(length), dtype=float)
+    f = np.vectorize(lambda x: amplitude * math.sin(2 * math.pi * frequency * x + phase) + y_offset)
+    values = f(values)
+
+    return TimeSeries.from_times_and_values(times, values)
+
+
+def white_noise_timeseries(length: int = 10, offset: str = 'D', mean: float = 0, std: float = 1,
+                           start_date: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
+    """
+    Creates a white noise TimeSeries by sampling a gaussian distribution with mean 'mean' and 
+    standard deviation 'std'. Each value represents an indipendent sample of the distribution.
+
+    :param mean: The mean of the gaussian distribution that is sampled at each step.
+    :param std: The standard deviation of the gaussian distribution that is sampled at each step.
+    :param length: The length of the returned TimeSeries.
+    :param offset: The time differene between two adjacent entries in the returned TimeSeries. A DateOffset alias is expected.
+    :param start_date: The time index of the first entry in the returned TimeSeries.
+    :return: A white noise TimeSeries created as indicated above.
+    """
+
+    times = pd.date_range(periods=length, freq=offset, start=start_date)
+    values = np.random.normal(mean, std, size=length)
+
+    return TimeSeries.from_times_and_values(times, values)
+
+
+def random_walk_timeseries(length: int = 10, offset: str = 'D', mean: float = 0, std: float = 1,
+                           start_date: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
+    """
+    Creates a random walk TimeSeries by sampling a gaussian distribution with mean 'mean' and 
+    standard deviation 'std'. The first value is one such random sample. Every subsequent value
+    is equal to the previous value plus a random sample.
+
+    :param mean: The mean of the gaussian distribution that is sampled at each step.
+    :param std: The standard deviation of the gaussian distribution that is sampled at each step.
+    :param length: The length of the returned TimeSeries.
+    :param offset: The time differene between two adjacent entries in the returned TimeSeries. A DateOffset alias is expected.
+    :param start_date: The time index of the first entry in the returned TimeSeries.
+    :return: A random walk TimeSeries created as indicated above.
+    """
+
+    times = pd.date_range(periods=length, freq=offset, start=start_date)
+    values = [np.random.normal(mean, std)]
```

Comment:
I think here there are probably ways to generate the random vector of steps once, and then take its cumulative sum in one go using numpy (which would likely be more efficient)